osbuild/meta.py

```diff
@@ -484,6 +486,46 @@ def load(cls, name):
         return cls(mod)
 
 
+class RunnerInfo:
+    """Information about a runner
+
+    Class that represents an actual available runner for a
+    specific distribution and version.
+    """
+
+    def __init__(self, distro: str, version: int, path: str) -> None:
+        self.distro = distro
+        self.version = version
+        self.path = path
+
+    @classmethod
+    def from_path(cls, path: str):
+        name = os.path.basename(path)
+        distro, version = cls.parse_name(name)
+        return cls(distro, version, path)
+
+    @staticmethod
+    def parse_name(name: str) -> Tuple[str, int]:
```

Comment:
In this case I feel a regex is quite a bit shorter than the manual parsing. `re.search(r"^(\w+?)(\d+)?$", "abdcdfxcavasdff1").groups()` and also verifies that a runner isn't named wrong somewhere.

Reply:
It is shorter, no question, but my eyes. ;) I would accept everything as the name that is not the trailing number though. I think I leave this for now as is, can always refactor later.